File: LeetCode/Easy/2200-Find_All_K-Distant_Indices_In_An_Array/2200-FindAllK-DistantIndicesInAnArray.js
/**
 * @param {number[]} nums
 * @param {number} key
 * @param {number} k
 * @return {number[]}
 */
var findKDistantIndices = function (nums, key, k) {
    let ans = [];
    let n = nums.length;

    let right = 0;
    let left = 0;

    for (let i = 0; i < n; i++) {
        if (nums[i] == key) {
            left = Math.max(right, i - k);
            right = Math.min(n - 1, i + k) + 1;
            for (let idx = left; idx < right; idx++) {
                ans.push(idx);
            }
        }
    }

    return ans
};

File: LeetCode/Easy/2200-Find_All_K-Distant_Indices_In_An_Array/2200-FindAllK-DistantIndicesInAnArray.cpp
class Solution {
public:
    vector<int> findKDistantIndices(vector<int>& nums, int key, int k) {
        vector<int> ans;
        int n = nums.size();

        int right = 0;
        int left = 0;

        for (int i = 0; i < n; i++) {
            if (nums[i] == key) {
                left = max(right, i - k);
                right = min(n - 1, i + k) + 1;
                for (int idx = left; idx < right; idx++) {
                    ans.push_back(idx);
                }
            }
        }

        return ans;
    }
};
